src/Integration.Vsix.UnitTests/CFamily/CFamilyIssueToAnalysisIssueConverterTests.cs

```diff
@@ -435,6 +435,24 @@ public void Convert_HasMessageParts_QualifiedFilePath(string originalPath, strin
             issue.Flows[0].Locations[0].FilePath.Should().Be(expectedPath);
         }
 
+        [TestMethod]
+        [Description("Regression test for https://github.com/SonarSource/sonarlint-visualstudio/issues/2557")]
+        [DataRow("", "")] // empty should not throw
+        [DataRow("a.txt", "a.txt")] // not-rooted should stay the same
+        [DataRow("c:\\a.txt", "c:\\a.txt")]
+        [DataRow("c:/a.txt", "c:\\a.txt")]
+        [DataRow("c:/a/b/c.txt", "c:\\a\\b\\c.txt")]
+        [DataRow("c:/a\\b/c.txt", "c:\\a\\b\\c.txt")]
+        public void Convert_FilePath_QualifiedFilePath(string originalPath, string expectedPath)
```

Comment:
We actually do the same thing for the secondary locations, test cases are copy-pasted from `Convert_HasMessageParts_QualifiedFilePath` above